internal/reader_with_retrys.go

```diff
@@ -0,0 +1,77 @@
+package internal
```

Comment:
we have the [ioextensions](https://github.com/wal-g/wal-g/tree/master/internal/ioextensions) package for io-related stuff, I guess this should be there